corehq/apps/sms/views.py

```diff
@@ -1039,11 +1050,15 @@ def backend_class_name(self):
     def ignored_fields(self):
         return [
             'give_other_domains_access',
+            'phone_numbers',
         ]
 
     @property
     @memoized
     def backend_class(self):
+        # For now, only allow superusers to create/edit non-Telerivet backends
+        if not self.is_superuser and self.backend_class_name != "TelerivetBackend":
```

Comment:
The comment seems to say something different from what the code is doing. Which is correct? The comment seems like it could mean either of the following:
```python
# non-superuser cannot edit any backends
# superuser can edit all except Telerivet backends
if not self.is_superuser or self.backend_class_name == "TelerivetBackend":
    raise Http404()
```
```python
# superuser is limited to non-Telerivet backends
# other users have full access to all backends
if self.is_superuser and self.backend_class_name == "TelerivetBackend":
    raise Http404()
```
I may be misunderstanding the meaning of raising 404. I assumed that 404 meant that the user could not edit that backend.